code/game/gamemodes/objective.dm

```diff
@@ -17,8 +17,20 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 
 /datum/objective/Destroy()
 	GLOB.all_objectives -= src
+	if(owner)
+		owner.objectives -= src
 	return ..()
 
+/**
+ * Will try to renew the objective to a valid state with another target.
+ * Returns the new found target or null if none were found
+ */
+/datum/objective/proc/on_target_loss()
+	if(owner?.current)
+		to_chat(owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+		SEND_SOUND(owner.current, 'sound/ambience/alarm4.ogg')
```

Comment:
I'm not 100% sure, but I think `playsound()` is preferred over this, just judging by how many more instances of it there are than `SEND_SOUND`.

Reply:
`playsound` will make it hearable for all others nearby as well. `playsound_local` is an option though which is used by the AI and traitor intro sound... oddly enough `SEND_SOUND` is used by the changeling intro sound.
I'll use `playsound_local` here in the same way the agents do it!